custom/enikshay/model_migration_sets/private_nikshay_notifications.py

```diff
@@ -0,0 +1,45 @@
+from casexml.apps.case.util import get_datetime_case_property_changed
+from custom.enikshay.const import (
+    ENROLLED_IN_PRIVATE,
+    REAL_DATASET_PROPERTY_VALUE,
+)
+
+
+class PrivateNikshayNotifiedDateSetter(object):
+    """Sets the date_private_nikshay_notification property for use in reports
+    """
+
+    def __init__(self, domain, person, episode):
+        self.domain = domain
+        self.person = person
+        self.episode = episode
+
+    def update_json(self):
+        if not self.should_update:
+            return {}
+
+        registered_datetime = get_datetime_case_property_changed(
+            self.episode, 'private_nikshay_registered', 'true',
+        )
+        if registered_datetime is not None:
+            return {
+                'date_private_nikshay_notification': str(registered_datetime.date())
+            }
+        else:
+            return {}
+
+    @property
+    def should_update(self):
+        if self.episode.get_case_property('date_private_nikshay_notification') is not None:
+            return False
+
+        if self.episode.get_case_property('private_nikshay_registered') != 'true':
+            return False
+
+        if self.episode.get_case_property(ENROLLED_IN_PRIVATE) != 'true':
+            return False
+
+        if self.person.get_case_property('dataset') != REAL_DATASET_PROPERTY_VALUE:
```

Comment:
Is this available for private app as well? @sheelio mentioned this was just for the public app during the change we made [here](https://github.com/dimagi/commcare-hq/blob/master/custom/enikshay/integrations/utils.py#L34)
If available,
Is it there for all cases and not only for ones after this was added?

Reply:
This is probably a redundant check here, but yes, I believe that case property is available in the private sector app.

- The Beneficiary Register (scoped to the CTD location) lists 160,003 entries
- Case search with "enrolled_in_private = true" --> 352,358
-  "enrolled_in_private" = "true" and "dataset = real" --> 158,440 entries, 
- "enrolled_in_private = true" and "dataset = test" --> 1,366 entries

i.e. private sector patients with dataset property = 159,806. 

However, the numbers don't match completely. Either there is some test data that's made it's way into the beneficiary register (which doesn't check the dataset property), or some cases were created before this property existed... 

Regardless, I'll remove this check as it sounds like it is inconsistent with our data, and any Nikshay Registered patients have a more robust check anyway.